lib/formatter.ex

```diff
@@ -0,0 +1,3 @@
+defmodule Elixometer.Formatter do
+  @callback format(String.t(), String.t()) :: [String.t()]
```

Comment:
You can give the parameters names in this typespec to make it a bit more readable:
```elixir
@type metric_type :: :counter | :gauge | :histograms | :spirals | :timer
@callback format(metric_type(), metric_name :: String.t()) :: [String.t()]
```
Also I think the specis incorrect, the first parameter is an atom instead of a string.